env0/resource_environment_test.go

```diff
@@ -182,6 +184,42 @@ func TestUnitEnvironmentResource(t *testing.T) {
 			})
 		})
 
+		t.Run("Import By Id", func(t *testing.T) {
+			testCase := resource.TestCase{
+				Steps: []resource.TestStep{
+					{
+						Config: createEnvironmentResourceConfig(environment),
+					},
+					{
+						ResourceName:            resourceNameImport,
+						ImportState:             true,
+						ImportStateId:           environment.Id,
+						ImportStateVerify:       true,
+						ImportStateVerifyIgnore: []string{"force_destroy"},
+					},
+				},
+			}
+
+			runUnitTest(t, testCase, func(mock *client.MockApiClientInterface) {
+				mock.EXPECT().Template(environment.LatestDeploymentLog.BlueprintId).Times(1).Return(template, nil)
+				mock.EXPECT().EnvironmentCreate(client.EnvironmentCreate{
+					Name:                       environment.Name,
+					ProjectId:                  environment.ProjectId,
+					WorkspaceName:              environment.WorkspaceName,
+					AutoDeployByCustomGlob:     autoDeployByCustomGlobDefault,
+					TerragruntWorkingDirectory: environment.TerragruntWorkingDirectory,
+					VcsCommandsAlias:           environment.VcsCommandsAlias,
+					DeployRequest: &client.DeployRequest{
+						BlueprintId: templateId,
+					},
+					IsRemoteBackend: &isRemoteBackendFalse,
+				}).Times(1).Return(environment, nil)
+				mock.EXPECT().Environment(environment.Id).Times(3).Return(environment, nil)
+				mock.EXPECT().EnvironmentDestroy(environment.Id).Times(1)
+				mock.EXPECT().ConfigurationVariablesByScope(client.ScopeEnvironment, environment.Id).Times(3).Return(client.ConfigurationChanges{}, nil)
```

Comment:
what is tested by these lines? the `times(3)` isn't really clear to me

Reply:
Terraform calls "Read" multiple times.
First time after creation.
Another time to check for drifts.
Another time for import.